scratch/copy-mk.py

```diff
@@ -89,34 +111,47 @@ async def multi_sensor_values(client: KATPortalClient, names: Sequence[str]) ->
 
 def delay_transfer_callback(
     msg_dict: dict,
-    delays_queue: asyncio.Queue[tuple[int, str, str]],
-) -> tuple:
-    """Transfer F-engine delays from MK subarray to MK+ F-engines.
+    delays_queue: asyncio.Queue[DelayQueueItem],
+) -> None:
+    """Receive delay sensor updates from katportal.
 
-    It currently puts a tuple on the `delays_queue` for each sensor update
-    in the format (loadmcnt, src_name, request_args). `request_args` is a
-    string formatted as:
+    Populate the `delays_queue` with an item containing the loadmcnt,
+    input_label and request_args from the delay sensor. `request_args`
+    is a string formatted as:
     - delay,delay_rate:phase,phase_rate
     """
     delay_sensor_value = msg_dict["msg_data"]["value"]
-    src_name: str = msg_dict["msg_data"]["name"].split("_")[-2]
     # (loadmcnt <ADC sample count when model was loaded>, delay <in seconds>,
     # delay-rate <unit-less or, seconds-per-second>, phase <radians>,
     # phase-rate <radians per second>)
-    loadmcnt, delay, delay_rate, phase, phase_rate = delay_sensor_value.replace(" ", "").split(",")
-    request_args = f"{delay},{delay_rate}:{phase},{phase_rate[:-1]}"
-    request_tuple = (int(loadmcnt[1:]), src_name, request_args)
+    loadmcnt, delay, delay_rate, phase, phase_rate = ast.literal_eval(delay_sensor_value)
+    request_args = f"{delay},{delay_rate}:{phase},{phase_rate}"
+    # Example sensor name: cbf_1_wide_antenna_channelised_voltage_m000h_delay
+    input_label: str = msg_dict["msg_data"]["name"].split("_")[-2]
+    delay_queue_item = DelayQueueItem(
+        loadmcnt=int(loadmcnt),
```

Comment:
Is the cast to int still needed? ast.literal_eval probably gives you an int to start with.

Reply:
You may very well be right. I can't test at the moment because nothing is running. I'll remove the cast anyway.